cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -391,13 +392,13 @@ protected void onFinish(HttpResponder responder, File uploadedFile) {
   private ProgramTerminator createProgramTerminator() {
     return new ProgramTerminator() {
       @Override
-      public void stop(Id.Program programId) throws Exception {
+      public void stop(ProgramId programId) throws Exception {
         switch (programId.getType()) {
           case FLOW:
             stopProgramIfRunning(programId);
             break;
           case WORKFLOW:
-            scheduler.deleteSchedules(programId, SchedulableProgramType.WORKFLOW);
+            scheduler.deleteSchedules(programId.toId(), SchedulableProgramType.WORKFLOW);
```

Comment:
if you make changes to the ``Scheduler`` APIs in this PR, does the size of the PR blow up?